### PR TITLE
[VPU] coverity fixes

### PR DESCRIPTION
**Pull request by KorA6**
### Tickets:
 - *69387*
 - *69388*

### CODE REVIEW

File: inference-engine/thirdparty/movidius/XLink/shared/src/XLinkDispatcher.c [hunk 1]
```diff
@@ -996,7 +996,13 @@ static int dispatcherClean(xLinkSchedulerState_t* curr)
 
         XLINK_RET_ERR_IF(pthread_mutex_lock(&(curr->queueMutex)) != 0, 1);
         postAndMarkEventServed(event);
-        XLINK_RET_ERR_IF(pthread_mutex_unlock(&(curr->queueMutex)) != 0, 1);
+        if (pthread_mutex_unlock(&(curr->queueMutex)) != 0) {
+            if (pthread_mutex_unlock(&clean_mutex) != 0) {
+                mvLog(MVLOG_ERROR, "Failed to unlock clean_mutex after failing to lock curr->queueMutex");
```
Comment (AntonDudchenko) [inline inference-engine/thirdparty/movidius/XLink/shared/src/XLinkDispatcher.c:1001]: Failed to unlock clean_mutex after failing to **_unlock_** curr->queueMutex

File: inference-engine/thirdparty/movidius/XLink/shared/src/XLinkDispatcher.c [hunk 2]
```diff
@@ -996,7 +996,13 @@ static int dispatcherClean(xLinkSchedulerState_t* curr)
...
+            }
+            mvLog(MVLOG_ERROR, "Failed to lock curr->queueMutex");
```
Comment (AntonDudchenko) [inline inference-engine/thirdparty/movidius/XLink/shared/src/XLinkDispatcher.c:1003]: The same

Comment (KorA6) [reply inference-engine/thirdparty/movidius/XLink/shared/src/XLinkDispatcher.c]: Done

File: inference-engine/thirdparty/movidius/XLink/shared/src/XLinkDispatcher.c (hunk 1 above)
Comment (KorA6) [reply inference-engine/thirdparty/movidius/XLink/shared/src/XLinkDispatcher.c]: Done

**Review by AntonDudchenko — APPROVED** [review]

**Review by DariaMityagina — APPROVED** [review]

**Pull request merged**